[master] salt.channel.server.handle_message: be more defensive

mattp- opened the pull request:
### What does this PR do?
it's possible _encrypt_private can throw depending on what key backend is used; so we should guard against any possible unhandled exceptions in this critical codepath. We already do this when wrapping the payload_handler, so we just move the other code in the same try.

I've also added some tests to cover different scenarios in handle_message as its untested atm.

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**
<!-- Please review the [test documentation](https://docs.saltproject.io/en/master/topics/tutorials/writing_tests.html) for details on how to implement tests into Salt's test suite. -->
- [ ] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
No

mattp- committed:
salt.channel.server.handle_message: be more defensive

it's possible _encrypt_private can throw depending on what key backend
is used; so we should guard against any possible unhandled exceptions in
this critical codepath. We already do this when wrapping the
payload_handler, so we just move the other code in the same try.

twangboy reviewed (changes requested):
Please add a changelog
There are also some failing tests

mattp- commented:
@twangboy will do, still looking at that. thanks

mattp- commented:
@twangboy I didn't change anything but pushed the changelog.md, it seems to be passing minus one test. would you mind a sanity check/merge? Thanks 👍

mattp- commented:
@dwoz sure, done

mattp- commented:
@twangboy woops, sorry about that. fixed

mattp- committed:
salt.channel.server: catch all exception, not just invalid key

also log stack on debug. we wan't this to be handled no matter what as
if somehow get_rsa_pub_key raises an exception that isn't of the
expected type, it fully hangs the mworker, essentially causing a denial
of service.

mattp- commented:
sure just let me know in the #develop discord and i'll rebase all the PR's
